core/services/ocr2/plugins/ocr2keeper/evm/registry.go

```diff
@@ -69,11 +102,23 @@ func NewEVMRegistryServiceV2_0(addr common.Address, client evm.Chain, mc *models
 		client:   client.Client(),
 		txHashes: make(map[string]bool),
 		registry: registry,
-		abi:      abi,
+		abi:      keeperRegistryABI,
 		active:   make(map[string]activeUpkeep),
-		packer:   &evmRegistryPackerV2_0{abi: abi},
+		packer:   &evmRegistryPackerV2_0{abi: keeperRegistryABI},
 		headFunc: func(types.BlockKey) {},
 		chLog:    make(chan logpoller.Log, 1000),
+		mercury: MercuryConfig{
+			cred:          mc,
+			abi:           mercuryLookupCompatibleABI,
+			upkeepCache:   upkeepInfoCache,
+			cooldownCache: cooldownCache,
+			apiErrCache:   apiErrCache,
+		},
+		hc: &http.Client{
+			// this timeout is a catch-all default value and should be less than delta round
+			// the actual timeout is determined by the context from plugin
```

Comment:
not sure if this is strictly needed? If so we should discuss further since plugin config can change without this changing

Reply:
only needed before we pass context into request. removed